pkgs/by-name/co/coredns/package.nix

```diff
@@ -53,7 +34,35 @@ buildGoModule rec {
 
   # Override the go-modules fetcher derivation to fetch plugins
   modBuildPhase = ''
-    for plugin in ${builtins.toString (attrsToPlugins externalPlugins)}; do echo $plugin >> plugin.cfg; done
+    cp plugin.cfg plugin.cfg.orig
+    ${
+      (lib.concatMapStringsSep "\n" (
+        plugin:
+        let
+          position = plugin.position or "end-of-file";
+          formatPlugin = { name, repo, ... }: "${name}:${repo}";
+        in
+        if position == "end-of-file" then
+          "echo ${formatPlugin plugin} >> plugin.cfg"
+        else if position == "start-of-file" then
+          "sed -i '1i ${formatPlugin plugin}' plugin.cfg"
+        else if lib.hasAttrByPath [ "before" ] position then
+          "sed -i '/^${position.before}:/i ${formatPlugin plugin}' plugin.cfg"
```

Comment:
does the sed invocation fail if the other plugin isn't found?

Reply:
I just tested this and it doesn't fail, it does nothing. I'll look into generating an error for this.

Reply:
Just pushed a change that stops the build if we're unable to insert a plugin before or after another plugin.